pkg/services/hcloud/server/server_test.go

```diff
@@ -339,3 +340,96 @@ var _ = Describe("Test ValidateLabels", func() {
 		}),
 	)
 })
+
+var _ = Describe("Test handleRateLimit", func() {
+	type testCaseHandleRateLimit struct {
+		hm           *infrav1.HCloudMachine
+		err          error
+		functionName string
+		errMsg       string
+		expectError  error
+		conditionSet bool
+	}
+
+	DescribeTable("Test handleRateLimit",
+		func(tc testCaseHandleRateLimit) {
+			err := handleRateLimit(tc.hm, tc.err, tc.functionName, tc.errMsg)
+			if tc.expectError != nil {
+				Expect(err).To(MatchError(tc.expectError))
+			} else {
+				Expect(err).To(BeNil())
+			}
+			if tc.conditionSet {
+				Expect(conditions.Has(tc.hm, infrav1.HetznerAPIReachableCondition)).To(BeTrue())
```

Comment:
you changed it to the worse here: You had a good check that checked that the condition is on false and with a speciifc reason.. that was correct. The check in the "else" condition you had before didn't make sense!